src/Database/QueryCompiler.php

```diff
@@ -153,7 +153,7 @@ protected function _buildSelectPart($parts, $query, $generator)
         $normalized = [];
         $parts = $this->_stringifyExpressions($parts, $generator);
         foreach ($parts as $k => $p) {
-            if (!is_numeric($k)) {
+            if ($query->autoAlias() && !is_numeric($k)) {
```

Comment:
As feared not the best solution here (see travis). I will try to find a better solution. Probably it's better to handle in `ORM/Query.php`.